.setup/install_submitty/setup_directories.sh

```diff
@@ -12,13 +14,13 @@ EOF
 ########################################################################################################################
 ########################################################################################################################
 # this script must be run by root or sudo
-if [[ "$UID" -ne "0" ]] ; then
+if [[ "${UID:?}" -ne "0" ]] ; then
```

Comment:
What's the purpose of the `:?` here?  Based on what the [docs](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) say, it looks like this isn't really useful because no error message is being provided here.